benchmarks/data_utils.py

```diff
@@ -0,0 +1,119 @@
+
+        # stack frames as channels, x shape is [2, img_width, img_height]
+        sample = torch.stack((first_frame, last_frame))
+
+       
+        if self.transform:
+            sample = self.transform(sample)
+            
+        return sample, target_class
+
+
+#TODO is this batchable, I think it is
+# TODO, return some indication that image is empty. 
+    #TODO - if it ends up empty, did protein move or dissapear?? how much do we handle at the moment
+#TODO also this is kinda slow if that becomes a problem.
+class RoiTransform:
+    """Pick a ROI based on pixel intensity. Choose the same coordinates for both/all frames"""
+
+    def __init__(self, window_size=(28,28), kernel_size=5, use_first_frame_values=True):
+        """
+        window_size = (width, height) of cropped section  
+        kernel_size = size of gaussian kernel used in selecting peak pixel values  
+        use_first_frame_values = if true, set ROI based off first frame, if false, set ROI based off last frame
+        """
+        self.width = window_size[0]
+        self.height = window_size[1]
+        self.kernel_size = kernel_size
+        self.use_first_frame = use_first_frame_values
+
+    def __call__(self, sample):
+        """
+        sample is [n_channels, img_width, img_height]
+        """
+
+        first_frame, last_frame = sample.split(1, dim=0)
+
+
+        if self.use_first_frame:
+            img_blurred = TF.gaussian_blur(first_frame, kernel_size=self.kernel_size) # TODO setting?
+        else:
+            img_blurred = TF.gaussian_blur(last_frame, kernel_size=self.kernel_size) # TODO setting?
+
+        # used the topk function in case we later decide to take multiple peaks
+        values, indices = torch.topk(img_blurred.flatten(), k=1)
+
+        """ 
+        Need to address the empty images (no mitochondria pictured) before train/test stage
+        """
+        # If first frame is empty, check the last frame for a signal
+        if values[0] == 0.0:
+            img_blurred = TF.gaussian_blur(last_frame, kernel_size=self.kernel_size) # TODO setting?
+            values, indices = torch.topk(img_blurred.flatten(), k=1)
+            if values[0] == 0.0:
+                print("NO DATA: maximum pixel value is {}, but this sample will still be trained on".format(values[0]))
+
+        np_idx = np.array(np.unravel_index(indices.numpy(), img_blurred.shape))
+
+        top = int(np_idx[1] - 1/2 * self.width)
+        left = int(np_idx[2] - 1/2 * self.height)
+        assert first_frame.shape == img_blurred.shape
+
+        # The crop is taken from original, unblurred image
+        first_cropped = torchvision.transforms.functional.crop(first_frame, top, left, self.width, self.height)
+        last_cropped = torchvision.transforms.functional.crop(last_frame, top, left, self.width, self.height)
+        cropped = torch.cat((first_cropped, last_cropped), 0)
+        # print("DEBUG: roi input size", cropped.size())
+
+        return cropped
```

Comment:
```suggestion
        return TF.crop(first_frame, top, left, self.width, self.height)
```

Reply:
This function is set up to take in 2 channel samples and output 2 channel crops. I don't think we want to return just one frame without bigger changes. I couldn't think of an easier way to share the crop coordinates between frames. I definitely should add better documentation.

Reply:
I changed this method significantly, the use of tensors might be clearer/better now